docs/design/2022-07-20-insert-support.md

```diff
@@ -0,0 +1,112 @@
+# TiSpark Design Documents
+
+- Author(s): TrafalgarRicardoLu
+- Tracking Issue: https://github.com/pingcap/tispark/issues/2462
+
+## Table of Contents
+- [Introduction](#introduction)
+- [Motivation or Background](#motivation-or-background)
+- [Goals](#goals)
+- [API](#api)
+- [Detailed Design](#detailed-design)
+  * [Add new capability](#add-new-capability)
+  * [Remove write options](#remove-write-options)
+  * [Implement V2 interface using V1 code](#implement-v2-interface-using-v1-code)
+  * [Compatible with different Spark versions](#compatible-with-different-spark-versions)
+  * [Why not V2 write model](#why-not-v2-write-model)
+- [Compatibility](#compatibility)
+- [Test Design](#test-design)
+  * [Functional Tests](#functional-tests)
+  * [Compatibility Tests](#compatibility-tests)
+- [References](#references)
+
+## Introduction
+
+New feature: TiSpark Insert SQL Feature
+
+## Motivation or Background
+
+TiSpark is a connector that provides read, write and delete support.
+But it only supports writing data using DataSource API, users can't use Spark SQL to write data.
+
+It's common for users to use SQL statements in the big-data workflow. We are going to support TiSpark to
+write using Spark SQL.
+
+## Goals
+**Must have**
+- Support write data by Spark SQL
+- Compatible with Spark 3.x
+
+**Nice to have**
+- Type convert compatible with the previous version
+- User doc
+
+## API
+Use spark SQL to write data in TiSpark.
+```
+// insert with spark sql
+spark.sql("insert into tidb_db.db.table values(x, x...)(y, y...)...")
+```
+
+## Detailed Design
+Here are the main steps:
+- Add new capability
+- remove write options
+- Implement V2 interface using V1 code
+- Compatible with different Spark versions
+
+### Add new capability
+Spark will check the capabilities of TiDBTable. If we don't add corresponding capability, Spark will throw an exception.
+
+### Remove write options
+Different from writing using DataSource API, Spark SQL can't get required information like tidb address and username.
+We use it to create a JDBC client to connect with TiDB, but we only need to use the JDBC client to update statics now.
+
+So we are going to remove the options check before writing and useless JDBC client code.
```

Comment:
Doesn't really get it. Does this affect DataSource API?

Reply:
It doesn't. Users can use DataSource API like before. Options can be indicate by `df.write.options(tidbOptions)` in DataSource API. But we can't get it in Spark SQL.

In fact, we only need it in our write path when `enableUpdateTableStatistics` is true. So we decide to remove useless code. If users really want enable it , they can set these information by `spark.conf.set()`.

As for DataSource API, user can still use it like before. But we don't use it if `enableUpdateTableStatistics` is false.

Reply:
you can add something like `DataSource API write will not be affected` to avoid the confusion